jax/_src/numpy/util.py

```diff
@@ -133,6 +133,39 @@ def _arraylike(x: ArrayLike) -> bool:
           hasattr(x, '__jax_array__') or np.isscalar(x))
 
 
+def _arraylike_asarray(x: Any) -> Array:
+  """Convert an array-like object to an array."""
+  if hasattr(x, '__jax_array__'):
+    x = x.__jax_array__()
+  elif isinstance(x, (bool, int, float, complex)):
+    x = dtypes.coerce_to_array(x)
+  return lax.asarray(x)
+
+
+@overload
+def ensure_arraylike(fun_name: str, /) -> tuple[()]: ...
+@overload
+def ensure_arraylike(fun_name: str, a1: Any, /) -> Array: ...
+@overload
+def ensure_arraylike(fun_name: str, a1: Any, a2: Any, /) -> tuple[Array, Array]: ...
+@overload
+def ensure_arraylike(fun_name: str, a1: Any, a2: Any, a3: Any, /) -> tuple[Array, Array, Array]: ...
+@overload
+def ensure_arraylike(fun_name: str, a1: Any, a2: Any, a3: Any, a4: Any, /, *args: Any) -> tuple[Array, ...]: ...
+def ensure_arraylike(fun_name: str, /, *args: Any) -> Array | tuple[Array, ...]:
+  """Check that arguments are arraylike and convert them to arrays."""
+  check_arraylike(fun_name, *args)
+  if len(args) == 1:
+    return _arraylike_asarray(args[0])  # pytype: disable=bad-return-type
+  return tuple(_arraylike_asarray(arg) for arg in args)  # pytype: disable=bad-return-type
+
+
+def ensure_arraylike_tuple(fun_name: str, tup: tuple[Any, ...]) -> tuple[Array, ...]:
+  """Check that argument elements are arraylike and convert to a tuple of arrays."""
+  check_arraylike(fun_name, *tup)
+  return tuple(_arraylike_asarray(arg) for arg in tup)
```

Comment:
I was confused about why this was needed (vs. just using `ensure_arraylike(name, *tup)`), and I think it's because `ensure_arraylike` won't return a tuple when there's only one input, right? Might be worth adding a comment?

Reply:
Done!